runtime/src/main/java/org/corfudb/runtime/view/TableRegistry.java

```diff
@@ -597,10 +607,58 @@ Table<K, V, M> openTable(@Nonnull final String namespace,
                 mapSupplier);
         tableMap.put(fullyQualifiedTableName, (Table<Message, Message, Message>) table);
 
+        resetIfTableTrimmed(namespace, tableName);
         registerTable(namespace, tableName, kClass, vClass, mClass, tableOptions);
         return table;
     }
 
+    /**
+     * If the table that is going to be registered, is not there in the registry table,
+     * but the address space seems to contain the stream map with no content, it means
+     * the table was trimmed without being checkpointing. If a trimmed table of this sort
+     * is found, append a HOLE to the stream to reset it.
+     *
+     * @param namespace Namespace of the table
+     * @param tableName Name of the table
+     */
+    private void resetIfTableTrimmed(String namespace, String tableName) {
+        TableName tableNameKey = TableName.newBuilder()
+                .setNamespace(namespace)
+                .setTableName(tableName)
+                .build();
+        CorfuRecord<TableDescriptors, TableMetadata> corfuRecord = this.registryTable.get(tableNameKey);
+        UUID streamId = CorfuRuntime.getStreamID(getFullyQualifiedTableName(namespace, tableName));
+        StreamAddressSpace streamAddressSpace = this.runtime.getSequencerView()
+                .getStreamAddressSpace(new StreamAddressRange(streamId, Address.MAX, Address.NON_ADDRESS));
+        if (streamAddressSpace.size() == 0 && streamAddressSpace.getTrimMark() != Address.NON_ADDRESS && corfuRecord == null) {
+            log.info("Found trimmed table that is re-opened. Reset table {}", getFullyQualifiedTableName(namespace, tableName));
+            resetTrimmedTable(namespace, tableName);
+        }
+    }
+
+    private void resetTrimmedTable(String namespace, String tableName) {
+        String table = TableRegistry.getFullyQualifiedTableName(namespace, tableName);
+        UUID streamId = CorfuRuntime.getStreamID(table);
+        CorfuTable corfuTable = runtime.getObjectsView()
+                .build()
+                .setStreamName(table)
+                .setTypeToken(new TypeToken<CorfuTable<Object, Object>>() {})
+                .setSerializer(Serializers.JSON)
+                .open();
+        CheckpointWriter<ICorfuTable<?,?>> cpw =
+                new CheckpointWriter<>(runtime, streamId, "corfu-store-resetTrimmedTable", corfuTable);
+
+        Token snapshot = cpw.forceNoOpEntry();
```

Comment:
If this snapshot is going to use stream tail then we can loose data from parallel transactions.
Please move this method and entire logic to within the TableRegistry's transaction which attempts to open the table and discovers that the `(oldRecord == null)`